scripts/test_screeps_runtime_monitor_tactical_response.py

```
@@ -446,6 +474,162 @@ def test_forced_room_discovery_preserves_overview_refs_for_summary_disambiguatio
+            with mock.patch.object(monitor.os, "kill") as kill:
+                monitor.signal_child_tree(1234, monitor.signal.SIGTERM)
+
+        kill.assert_called_once_with(1234, monitor.signal.SIGTERM)
+
+    def test_alert_timeout_returns_structured_diagnostic_json(self) -> None:
+        def stalled_alert(_args: object) -> int:
+            monitor.time.sleep(1)
+            return 0
+
+        with mock.patch.dict(monitor.os.environ, {"SCREEPS_AUTH_TOKEN": "abcdef123456"}, clear=True):
+            with mock.patch.object(monitor, "command_alert", new=stalled_alert):
```

Ruff (0.15.15)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)